mllib/src/test/scala/org/apache/spark/ml/param/ParamsSuite.scala

```diff
@@ -349,6 +352,41 @@ class ParamsSuite extends SparkFunSuite {
     val t3 = t.copy(ParamMap(t.maxIter -> 20))
     assert(t3.isSet(t3.maxIter))
   }
+
+  test("Filtering ParamMap") {
+    val params1 = new MyParams("my_params1")
+    val params2 = new MyParams("my_params2")
+    val paramMap = ParamMap(
+      params1.intParam -> 1,
+      params2.intParam -> 1,
+      params1.doubleParam -> 0.2,
+      params2.doubleParam -> 0.2)
+    val filteredParamMap = paramMap.filter(params1)
+
+    assert(filteredParamMap.size === 2)
+    filteredParamMap.toSeq.foreach {
+      case ParamPair(p, _) =>
+        assert(p.parent === params1.uid)
+    }
+
+    // At the previous implementation of ParamMap#filter,
+    // mutable.Map#filterKeys was used internally but
+    // the return type of the method is not serializable (see SI-6654).
+    // Now mutable.Map#filter is used instead of filterKeys and the return type is serializable.
+    // So let's ensure serializability.
+    val objOut = new ObjectOutputStream(new ByteArrayOutputStream())
+    try {
+      objOut.writeObject(filteredParamMap)
+    } catch {
+      case _: NotSerializableException =>
+        fail("The field of ParamMap 'map' may not be serializable. " +
+          "See SI-6654 and the implementation of ParamMap#filter")
+      case e: Exception =>
```

Comment:
This isn't necessary; an exception fails the test already. Really, same for the not-serializable exception. You don't need to close a stream that is just a byte-array output stream. Still it's all fine. Merge whenever you like.